test/utils/Create2.test.js

```diff
@@ -40,23 +40,48 @@ describe('Create2', function () {
     const offChainComputed =
       computeCreate2Address(saltHex, ERC20.bytecode, this.factory.address);
     await this.factory
-      .deploy(saltHex, ERC20.bytecode, { from: deployerAccount });
+      .deploy(0, saltHex, ERC20.bytecode, { from: deployerAccount });
     expect(ERC20.bytecode).to.include((await web3.eth.getCode(offChainComputed)).slice(2));
   });
 
   it('should deploy a ERC20Mock with correct balances', async function () {
     const offChainComputed =
       computeCreate2Address(saltHex, constructorByteCode, this.factory.address);
     await this.factory
-      .deploy(saltHex, constructorByteCode, { from: deployerAccount });
+      .deploy(0, saltHex, constructorByteCode, { from: deployerAccount });
     const erc20 = await ERC20Mock.at(offChainComputed);
     expect(await erc20.balanceOf(deployerAccount)).to.be.bignumber.equal(new BN(100));
   });
 
+  it('should deploy a contract with funds deposited in the factory', async function () {
+    const deposit = ether('2');
+    await send.ether(deployerAccount, this.factory.address, deposit);
+    expect(await balance.current(this.factory.address)).to.be.bignumber.equal(deposit);
+
+    const onChainComputed = await this.factory
+      .computeAddressWithDeployer(saltHex, web3.utils.keccak256(constructorByteCode), deployerAccount);
+
+    const amount = new BN(0);
```

Comment:
Alright so I double checked that the create2 parameters are correct, and that the mock contract has access to the ETH but whenever you set amount to a wei > 0 you get the following revert: Create2: Failed on deploy. Which implies that the opcode call is failing do I to mine a block before calling deploy on the test chain? I don't think that's the issue but I'm unsure of what exactly is wrong. Double checked that the parameter list is correct i.e. that 0x20 -> little endian is 32 etc